client/src/containers/filters/years-range/component.tsx

```diff
@@ -41,11 +41,13 @@ export const YearsRangeFilter: React.FC<YearsRangeFilterProps> = ({
   useEffect(() => {
     if (!years.length) return;
     setStartYearOptions(
-      years?.map((year) => ({
-        label: year.toString(),
-        value: year,
-        disabled: endYear <= year + (yearsGap - 1),
-      })),
+      years
+        ?.map((year) => ({
+          label: year.toString(),
+          value: year,
+          disabled: endYear <= year + (yearsGap - 1),
+        }))
+        .filter((y) => y.value <= lastYearWithData),
```

Comment:
```suggestion
      years
        ?.filter((year) => year <= lastYearWithData)
        .map((year) => ({
          label: year.toString(),
          value: year,
          disabled: endYear <= year + (yearsGap - 1),
        })),
```
better to `filter` before apply a `map`